plugin/code_actions.py

```diff
@@ -253,11 +252,14 @@ def _handle_response_async(self, responses: CodeActionsByConfigName) -> None:
         if self._cancelled:
             return
         document_version = self._view.change_count()
-        tasks = []
+        tasks = []  # type: List[Promise]
         for config_name, code_actions in responses.items():
             if code_actions:
-                for code_action in code_actions:
-                    tasks.append(run_code_action_or_command(self._view, config_name, code_action))
+                for session in sessions_for_view(self._view, 'codeActionProvider'):
+                    if session.config.name == config_name:
+                        for code_action in code_actions:
+                            tasks.append(session.run_code_action(code_action))
+                    break
```

Comment:
This might not be correct, any insights @rchl ?

Reply:
Apart from the misplaced `break` that you've corrected later, it's looking good to me.